projects/igniteui-angular/src/lib/grid/grid.component.ts

```diff
@@ -3722,16 +3722,24 @@ export class IgxGridComponent implements OnInit, OnDestroy, AfterContentInit, Af
 
         let size;
```

Comment:
I would remove the whole if/else block here and simply substitute it with:

```typescript
const size = directive.getItemCountInView();
```
since the new method determines the item count for both rows.